docs/hoon/library/2k.md

```diff
@@ -49,128 +52,119 @@ Source
         a
-      ::
 
 
-Examples
---------
+#### Examples
 
     > `(qeu tape)`["a" ~ "b" ~ "c" ~ "d" ~ "e" ~ "f" ~ "g" ~ ~]
     {"a" "b" "c" "d" "e" "f" "g"}
+
     > `*`["a" ~ "b" ~ "c" ~ "d" ~ "e" ~ "f" ~ "g" ~ ~]
     [[97 0] 0 [98 0] 0 [99 0] 0 [100 0] 0 [101 0] 0 [102 0] 0 [103 0] 0 0]
+
     > ~(bal to `(qeu tape)`["a" ~ "b" ~ "c" ~ "d" ~ "e" ~ "f" ~ "g" ~ ~])
-    {"a" "b" "c" "d" "e" "f" "g"}
+    [n=[i='d' t=""] l={"a" "b" "c"} r={"e" "f" "g"}]
+
     > `*`~(bal to `(qeu tape)`["a" ~ "b" ~ "c" ~ "d" ~ "e" ~ "f" ~ "g" ~ ~])
     [[100 0] [[99 0] [[98 0] [[97 0] 0 0] 0] 0] [101 0] 0 [102 0] 0 [103 0] 0 0]
 
-------------------------------------------------------------------------
-
+***
 ### `+-dep:to`
 
 Maximum Depth
 
-Produces the maximum depth of leaves (r.a and l.a) in queue `a`.
+Produces the maximum depth of leaves (`r.a` and `l.a`) in queue `a`.
 
-Accepts
--------
+#### Accepts
 
 `a` is a queue.
 
-Produces
---------
+#### Produces
 
 An atom.
 
-Source
-------
+#### Source
 
-      +-  dep                                               ::  max depth of queue
+      +-  dep                                              
         |-  ^-  @
         ?~  a  0
         +((max $(a l.a) $(a r.a)))
-      ::
 
 
-Examples
---------
+#### Examples
 
     > =a (~(gas to `(qeu @)`~) `(list @)`[1 2 3 4 5 6 7 ~])
     > ~(dep to a)
     4
+
     > =a (~(gas to `(qeu @)`~) `(list @)`[1 2 3 4 ~])
     > ~(dep to a)
     3
+
     > =a (~(gas to `(qeu @)`~) `(list @)`[1 2 ~])
     > ~(dep to a)
     2
+
     > ~(dep to `(qeu tape)`["a" ~ "b" ~ "c" ~ "d" ~ "e" ~ "f" ~ "g" ~ ~])
     7
+
     > ~(dep to ~(bal to `(qeu tape)`["a" ~ "b" ~ "c" ~ "d" ~ "e" ~ "f" ~ "g" ~ ~]))
     4
 
-------------------------------------------------------------------------
-
-### `+-gas`
+***
+### `+-gas:to`
 
 Push list
 
-Push all elements of `++list` `b` into the `++queue`.
+Push all elements of list `b` into the queue `a`.
 
-Accepts
--------
+#### Accepts
 
 `a` is a queue.
 
 `b` is a list.
 
-Produces
---------
+#### Produces
 
 A queue.
 
-Source
-------
+#### Source
+
+      +-  gas                                             
+        |=  b=(list _?>(?=(^ a) n.a))
+        |-  ^+  a
+        ?~(b a $(b t.b, a (put i.b)))
 
-    +-  gas                                               ::  insert list to queue
-      |=  b=(list ,_?>(?=(^ a) n.a))
-      |-  ^+  a
-      ?~(b a $(b t.b, a (put(+< a) i.b)))
-    ::
 
-Examples
---------
+#### Examples
 
-    > (~(gas to `(qeu ,@)`~) `(list ,@)`[1 2 3 ~])
+    > `(qeu)`(~(gas to `(qeu @)`~) [1 2 3 ~])
     {3 2 1}
-    > =a (~(gas to `(qeu ,@)`~) `(list ,@)`[1 2 3 ~])
-    > =b `(list ,@)`[4 5 6 ~]
-    > (~(gas to a) b)
-    {6 5 4 3 2 1}
 
-------------------------------------------------------------------------
+    > =a (~(gas to `(qeu @)`~) [1 2 3 ~])
+    > =b [4 5 6 ~]
+    > `(qeu)`(~(gas to a) b)
+    {6 5 4 3 2 1}
 
+***
 ### `+-get:to`
 
-Pop
+Head-tail pair
```

Comment:
I don't understand what 'head' means here. I assume it means the element that was inserted first? I think we should document that this is a FIFO queue, as opposed to some other kind of queue.